### PR TITLE
ceph: report error on timeout

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

If the healthcheck goroutine cannot check for the ceph status/times out
(if the mons are not in quorum) this should still be reflected in the
CRD status field.

Example:

```
status:
  ceph:
    details:
      error:
        message: 'failed to get status. . timed out: exit status 1'
        severity: Urgent
    health: HEALTH_ERR
    lastChanged: "2020-07-20T14:11:05Z"
    lastChecked: "2020-07-20T14:11:05Z"
    previousHealth: HEALTH_WARN
```

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/cluster/cephstatus.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ func (c *cephStatusChecker) checkStatus() {
 	status, err = cephclient.StatusWithUser(c.context, c.clusterInfo)
 	if err != nil {
 		logger.Errorf("failed to get ceph status. %v", err)
+		if err := c.updateCephStatus(cephStatusOnError(err.Error())); err != nil {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/cephstatus.go:109]: What is the `err` when the status request fails? We likely need to wrap it with something more descriptive.

Comment (leseb) [reply pkg/operator/ceph/cluster/cephstatus.go:109]: `message: 'failed to get status. . timed out: exit status 1'`

Comment (leseb) [reply pkg/operator/ceph/cluster/cephstatus.go:109]: Isn't that enough?

Comment (travisn) [reply pkg/operator/ceph/cluster/cephstatus.go:109]: Ah, the `StatusWithUser()` already wrapped it with a decent message. What's up with the extra `.`? It doesn't look very good.

Comment (leseb) [reply pkg/operator/ceph/cluster/cephstatus.go:109]: Yes, I know, it's just that the buffer is empty... I can fix that.

Comment (leseb) [reply pkg/operator/ceph/cluster/cephstatus.go:109]: Fixed

File: pkg/daemon/ceph/client/status.go [hunk 2]
```diff
@@ -164,7 +166,11 @@ func StatusWithUser(context *clusterd.Context, clusterInfo *ClusterInfo) (CephSt
 
 	buf, err := context.Executor.ExecuteCommandWithOutput(command, args...)
 	if err != nil {
-		return CephStatus{}, errors.Wrapf(err, "failed to get status. %s", string(buf))
+		errMessage := "failed to get ceph status"
+		if buf != "" {
+			errMessage = fmt.Sprintf("%s. %s", errMessage, string(buf))
```
Comment (travisn) [inline pkg/daemon/ceph/client/status.go:171]: if `buf != ""`, the errMessage will be included twice. What about returning the error here so it doesn't also get wrapped on the next line?

Comment (travisn) [reply pkg/daemon/ceph/client/status.go:171]: Wait, should we just log the `buf` if not empty? It doesn't seem necessary for the error being returned.

Comment (leseb) [reply pkg/daemon/ceph/client/status.go]: Both have been useful so far.

**Pull request merged**